Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs

```diff
@@ -25,29 +26,72 @@ public class PressableButton : MonoBehaviour, IMixedRealityTouchHandler
 
         [SerializeField]
         [Header("Press Settings")]
+        [Tooltip("The offset at which pushing starts.")]
+        private float startPushDistance = 0.0f;
+        public float StartPushDistance { get => startPushDistance; set => startPushDistance = value; }
+
+        [SerializeField]
         [Tooltip("Maximum push distance")]
         private float maxPushDistance = 0.2f;
+        public float MaxPushDistance { get => maxPushDistance; set => maxPushDistance = value; }
 
         [SerializeField]
         [FormerlySerializedAs("minPressDepth")]
         [Tooltip("Distance the button must be pushed until it is considered pressed.")]
         private float pressDistance = 0.02f;
+        public float PressDistance { get => pressDistance; set => pressDistance = value; }
 
         [SerializeField]
         [FormerlySerializedAs("withdrawActivationAmount")]
         [Tooltip("Withdraw amount needed to transition from Pressed to Released.")]
         private float releaseDistanceDelta = 0.01f;
+        public float ReleaseDistanceDelta { get => releaseDistanceDelta; set => releaseDistanceDelta = value; }
 
         [SerializeField]
-        [Tooltip("Speed of the object movement on release.")]
-        private float returnRate = 25.0f;
+        [Tooltip("Speed for retracting the moving button visuals on release.")]
+        [FormerlySerializedAs("returnRate")]
+        private float returnSpeed = 25.0f;
 
         [SerializeField]
         [Tooltip("Ensures that the button can only be pushed from the front. Touching the button from the back or side is prevented.")]
         private bool enforceFrontPush = true;
 
-        [Header("Position markers")]
-        [Tooltip("Used to mark where button movement begins. If null, it will be automatically generated.")]
+        public enum SpaceMode
+        {
+            World,
+            Local
+        }
+
+        [SerializeField]
+        [HideInInspector]
```

Comment:
Why do you need to both serialize the field and hide it in the inspector?

Reply:
we want to store the value but we don't want it to be visible in the inspector because there's a conversion button for changing the mode.  I wanted it to be more visible that we're actually converting all the distances on switching the mode, that's why i don't directly expose the property and have that button instead.